lib/system.nim

```diff
@@ -3126,3 +3126,6 @@ export io
 
 when not defined(createNimHcr) and not defined(nimscript):
   include nimhcr
+
+when not defined(gcDestructors):
+  proc prepareStrMutation*(s: var string) {.inline.} = discard
```

Comment:
```suggestion
  proc prepareStrMutation*(s: var string) {.inline.} =
    ## String literals(i.e. "abc", etc) in ARC/ORC mode are "copy on write",
    ## therefore you should call `prepareStrMutation` before modifying the strings.
    discard
```

Reply:
"before modifying the strings via `addr`"

Reply:
this definitely needs a `runnableExamples`

Reply:
ok

Reply:
=> https://github.com/nim-lang/Nim/pull/17227